config/charset.c

```diff
@@ -70,3 +70,39 @@ int charset_validator(const struct ConfigSet *cs, const struct ConfigDef *cdef,
   FREE(&s);
   return rc;
 }
+
+/**
+ * charset_validator_slist - Validate the "charset" config variable - Implements ConfigDef::validator() - @ingroup cfg_def_validator
+ * This is a version for charset options is Slist.
+ */
+int charset_validator_slist(const struct ConfigSet *cs, const struct ConfigDef *cdef,
+                            intptr_t value, struct Buffer *err)
+{
+  if (value == 0)
+    return CSR_SUCCESS;
+
+  const struct Slist *list = (const struct Slist *) value;
+
+  if ((cdef->type & DT_CHARSET_SINGLE) && list->count > 1)
+  {
+    mutt_buffer_printf(err, _("'charset' must contain exactly one character set name"));
+    return CSR_ERR_INVALID;
+  }
```

Comment:
We'll probably only apply the `DT_CHARSET_SINGLE` to variables of type `DT_STRING`.
So we probably don't need this section.

Reply:
Right, that makes sense, if it's single, it probably shouldn't be a list.